README.md

```diff
@@ -119,6 +119,18 @@ $ ls build/pkg/txz/
 
 For integration with PyTorch, follow the dockerfile in this repo. It has an example for how to replace default NCCL with MSCCL.
 
+## NPKit Integration
+
+MSCCL integrates [NPKit](https://github.com/microsoft/npkit), a profiler framework that enables collecting fine-grained trace events in MSCCL components, especially in giant collective GPU kernels.
+
+Please check [NPKit sample workflow for MSCCL](https://github.com/microsoft/NPKit/tree/main/msccl_samples) as a fully automated usage example. It also provides good templates for the following manual instructions.
+
+To manually build MSCCL with NPKit enabled, pass `-DNPKIT_FLAGS="-DENABLE_NPKIT -DENABLE_NPKIT_...(other NPKit compile-time switches)"` with cmake command. All NPKit compile-time switches are declared in the MSCCL code base as macros with prefix `ENABLE_NPKIT_`, and they control which information will be collected. Also note that currently NPKit only supports collecting non-overlapped events on GPU, and `-DNPKIT_FLAGS` should follow this rule.
+
+To manually run MSCCL with NPKit enabled, environment variable `NPKIT_DUMP_DIR` needs to be set as the NPKit event dump directory. Also note that currently NPKit only supports 1 GPU per process.
+
+To manually analyze NPKit dump results, please leverage [npkit_trace_generator.py](https://github.com/microsoft/NPKit/blob/main/msccl_samples/npkit_trace_generator.py).
```

Comment:
a description of the meta data for each event would be nice. Is it in NPKit repo?